aws/s3_test.go

```diff
@@ -458,3 +462,62 @@ func TestNukeS3Bucket(t *testing.T) {
 		})
 	}
 }
+
+// TestFilterS3BucketArgs represents arguments for TestFilterS3Bucket_Config
+type TestFilterS3BucketArgs struct {
+	configFilePath string
+	matches        []string
+}
+
+func bucketNamesForConfigTests() []string {
+	return []string{
+		"alb-alb-123456-access-logs-" + S3TestGenBucketName(),
+		"alb-alb-234567-access-logs-" + S3TestGenBucketName(),
+		"tonico-prod-alb-access-logs-" + S3TestGenBucketName(),
+		"prod-alb-public-access-logs-" + S3TestGenBucketName(),
+		"stage-alb-internal-access-logs-" + S3TestGenBucketName(),
+		"stage-alb-public-access-logs-" + S3TestGenBucketName(),
+	}
+}
+
+// TestFilterS3Bucket_Config tests listing only S3 buckets that match config file
+func TestFilterS3Bucket_Config(t *testing.T) {
+	awsParams, err := newS3TestAWSParams()
+	require.NoError(t, err, "Failed to setup AWS params")
+
+	// Create test buckets
+	var bucketTags []map[string]string
+	bucketNames := bucketNamesForConfigTests()
+	for _, bucketName := range bucketNames {
+		err = S3TestCreateBucket(awsParams.svc, bucketName, bucketTags, false)
+		require.NoErrorf(t, err, "Failed to create test bucket - %s", bucketName)
+	}
+
+	type testCaseStruct struct {
+		name string
+		args TestFilterS3BucketArgs
+	}
+
+	filterMatches := make([]string, len(bucketNames))
+	copy(filterMatches, bucketNames)
+
+	testCase := testCaseStruct{
+		"IncludeAndExclude",
+		TestFilterS3BucketArgs{
+			configFilePath: "../config/mocks/s3_filter_names.yaml",
+			matches:        filterMatches[0:3],
+		},
+	}
+
+	var configObj *config.Config
+	configObj, err = config.GetConfig(testCase.args.configFilePath)
+
+	// Verify that only filtered buckets are listed
+	bucketNamesPerRegion, err := getAllS3Buckets(awsParams.awsSession, time.Now().Add(1*time.Hour), []string{awsParams.region}, "", 100, *configObj)
+	require.NoError(t, err, "Failed to list S3 Buckets")
+	require.Equal(t, len(testCase.args.matches), len(bucketNamesPerRegion[awsParams.region]))
+	require.Subset(t, aws.StringValueSlice(bucketNamesPerRegion[awsParams.region]), testCase.args.matches)
+
+	// TODO: this is happening too early
+	defer nukeAllS3Buckets(awsParams.awsSession, aws.StringSlice(bucketNames), 1000)
```

Comment:
This actually needs to be much earlier in the test. As it is now, if any of the `require.XXX` checks above fail, then it'll never even get to the `defer`, and this cleanup will never happen.

As for it happening too early, what makes you think so? What does the log output look like for the test?

Reply:
I'm just rerunning this again, and it's nuking just fine! Disregard--maybe something else fixed it.